specifyweb/frontend/js_src/lib/userinfo.js

```diff
@@ -2,12 +2,36 @@
 
 var _              = require('underscore');
 var initialContext = require('./initialcontext.js');
+const commonText = require('./localization/common').default;
+const $ = require('jquery');
 
     var userInfo = {};
     initialContext.load('user.json', function(data) {
         _.extend(userInfo, data, {
-            isReadOnly:  !_(['Manager', 'FullAccess']).contains(data.usertype)
+            isReadOnly: !_(['Manager', 'FullAccess']).contains(data.usertype)
         });
+    }, (error)=>{
```

Comment:
If you set the `agent` field to null as described above then you can check for that here instead of just assuming that BadRequest implies there was no agent.